arch/eventloop_posix_tcp.c

```diff
@@ -778,7 +779,42 @@ TCP_openActiveConnection(UA_ConnectionManager *cm, const UA_KeyValueMap *params,
                        "TCP\t| Lookup of %s failed (%s)",
                        hostname, gai_strerror(error));
 #endif
+
+#if defined(UA_ENABLE_DISCOVERY) && defined(UA_ENABLE_DISCOVERY_MULTICAST)
+        char *domain = ".local";
+        size_t len = strlen(domain);
```

Comment:
It might be simpler to use a static 256-byte buffer for the domain rewriting.
And test for the length instead of mallocing.